### DISCUSSION TITLE
Can you filter Pull Requests by merge conflicts or by the file changed?

### DISCUSSION THREAD

**Discussion by johnkennedy9147** [community/community#21777] [New to GitHub]
<p>Hi</p>
<p>I havent been able to find anything in the documentation nor in a general web search.</p>
<p>Is it possible to filter Pull Requests by either of these:</p>
<p>Pull requests which either have or do not have merge conflicts.</p>
<p>Filter by the file which has been changed, ideally allowing a directory structure to be included in the filter.</p>
<p>thanks</p>

**Comment by mpboom** [comment]
<p>If it isn’t in this document: <a href="https://help.github.com/articles/using-search-to-filter-issues-and-pull-requests/" rel="nofollow noopener">https://help.github.com/articles/using-search-to-filter-issues-and-pull-requests/</a> it’s not possible.</p>
<p>You could submit it as a feature request and you can always use labels to manually organize issues by properties such as merge conflicts.</p>

**Reply by rusackas** [reply]
That's not actually true though... for example, the doc shows you can filter for `is:draft` but doesn't cover the exclusion operator at all, `-` to search for things like `-is:draft`. The doc doesn't cover other filters as well, such as `draft:false` or `draft:true` which do seem to work, and indicate that there are any number of undocumented filters. 

(side note... sorry to pick on a 5 year old comment, but this stuff still surfaces in google searches. Nice work on the SEO! ;)

Ninja edit: the `-` operator _is_ covered on this page: https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests

There's a lot of overlap between these two pages, and they could probably be consolidated, but that's another discussion entirely ;)

**Reply by mpboom** [reply]
You're right. Though at the time of writing, the linked page _did_ cover all available filters, some have been added in more recent years.

For a complete list of possibilities you should indeed be able to refer to the page you linked ([Searching issues and pull requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests)). As far as I'm aware, all options/filters are listed there.

As for the original question, I believe that as of today, it still is not possible to search by merge conflicts or specific files changed.

**Comment by johnkennedy9147** [comment]
<p>Hi, thanks I read that already, and every bit of documentation I could find on searching or filtering.</p>
<p>Doesnt seem to be an existing feature.</p>
<p>Labelling is great but when working on a repo with 3000+ open PRs finding what you are looking for is difficult to impossible.</p>

**Comment by angela-crist** [comment, marked as answer]
<p>Hi <a class="mention" href="/johnkennedy9147">@johnkennedy9147</a>,</p>
<p>It would be great if you could submit a feature requests via our contact page (<a href="http://github.com/contact" rel="nofollow noopener">github.com/contact</a>). GitHub Staff working in the Forum will see and respond to those requests there.</p>
<p>That said, I do want to share that we are working on a new tool and process for submitting these sorts of things for the Community Forum (and for <a href="http://github.com" rel="nofollow noopener">github.com</a>). This is still in the investigative stage, and I don’t have a timeline to share, but we are planning on improving this process.</p>
<p>We also recognize the desire for feature requests to be visible to other users, commented on, and even voted on. These are all things that we are taking into consideration when looking at implementing a new process.</p>
<p>Thank you so much for the feedback! I encourage you to send your feature request via the contact form linked above.</p>
<p>Please let me know if you have any other questions.</p>
<p>Cheers!</p>

**Comment by johnkennedy9147** [comment]
<p>Hi, thanks.</p>
<p>I did this earlier today and they already got back to me and advised they have added my suggestions to the Feature request list.</p>

**Comment by pdaire** [comment]
<p>Hi. </p>
<p>Is there any updates in this regard?</p>
<p>thank you in advanced</p>

**Comment by anurag-singh-df** [comment]
<p>Do we have this now ?</p>

**Comment by nilesh-godhani** [comment]
Filtering all PRs that gone in conflict is very much needed. Is this requirement is prioritized?

**Comment by cfriedt** [comment]
#want

**Comment by brunolm** [comment]
No — GitHub's search syntax has no qualifier for merge-conflict status. I checked the current docs qualifier list and community discussions: mergeability (`mergeable: CONFLICTING/MERGEABLE`) is only exposed through the GraphQL API, not the search index, so there's nothing you can append to that URL to exclude conflicted PRs. It's a long-standing feature request ([discussion #21777](https://github.com/orgs/community/discussions/21777), [#23152](https://github.com/orgs/community/discussions/23152)).

Two practical workarounds:

**1. `gh` CLI with GraphQL-backed field** — `gh pr list` (and `gh search prs`... actually only `gh pr list`) can expose `mergeable`:

```powershell
gh pr list -R group/repo --state open --draft=false --base main `
  --json number,title,author,mergeable,updatedAt,reviewRequests `
  --jq '.[] | select(.mergeable != "CONFLICTING" and .author.login != "brunolm") | "#\(.number) \(.title)"'
```

You'd have to replicate the `user-review-requested:@me` part with an extra filter on `reviewRequests`, but it gets you the conflict exclusion the web UI can't.

**2. Label automation** — a scheduled Action (e.g. the "Merge Conflict Finder"/`eps1lon/actions-label-merge-conflict` style actions) that stamps a `has-conflicts` label on dirty PRs; then your saved search just adds `-label:has-conflicts`. This is the only way to make the *browser URL itself* do the filtering.

If you mainly use that URL as a "what should I review" dashboard, option 2 is the one that keeps it a plain link; option 1 is zero-setup if a terminal command is acceptable.

Sources: [GitHub docs — Searching issues and pull requests](https://docs.github.com/en/search-github/searching-on-github/searching-issues-and-pull-requests), [community discussion #21777](https://github.com/orgs/community/discussions/21777), [community discussion #23152](https://github.com/orgs/community/discussions/23152)
